.github/workflows/ats.publish.yml

```
@@ -0,0 +1,160 @@
+name: ATS Publish
+
+on:
+  # Manual trigger with dry-run option
+  workflow_dispatch:
+    inputs:
+      dry-run-enabled:
+        description: 'Run npm publish with dry-run flag'
+        required: false
+        type: boolean
+        default: false
+
+  # Release published trigger (GitHub release creation) - only for ATS releases (tags ending with -ats)
+  release:
+    types:
+      - published
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  contracts:
+    name: Publish ATS Contracts
+    runs-on: token-studio-linux-large
+    # Only run if manual trigger OR release tag contains 'ats'/'ATS'
+    if: ${{ github.event_name == 'workflow_dispatch' || contains(github.ref_name, 'ats') || contains(github.ref_name, 'ATS') }}
```

This would match any branch or tag including the words:
- stats
- formats
- chats
- nats
- beats
- strats
- statsd
- automats

Assuming the branch/tag format is `<issue_no>-(ats|ATS)-<descriptive-branch-name>` then we should be doing something like this:

```suggestion
    if: ${{ github.event_name == 'workflow_dispatch' || contains(github.ref_name, '-ats-') }}
```

[Github also ignores case when doing comparisons](https://docs.github.com/en/actions/reference/workflows-and-actions/expressions#functions) so the second `contains` check is redundant and should be removed.